da/celestia/mock/server.go

```diff
@@ -70,137 +100,109 @@ func (s *Server) Stop() {
 
 func (s *Server) getHandler() http.Handler {
 	mux := mux2.NewRouter()
-	mux.HandleFunc("/submit_pfb", s.submit).Methods(http.MethodPost)
-	mux.HandleFunc("/namespaced_shares/{namespace}/height/{height}", s.shares).Methods(http.MethodGet)
-	mux.HandleFunc("/namespaced_data/{namespace}/height/{height}", s.data).Methods(http.MethodGet)
+	mux.HandleFunc("/", s.rpc).Methods(http.MethodPost)
 
 	return mux
 }
 
-func (s *Server) submit(w http.ResponseWriter, r *http.Request) {
-	req := cnc.SubmitPFBRequest{}
-	err := json.NewDecoder(r.Body).Decode(&req)
+func (s *Server) parseParams(w http.ResponseWriter, req request) []interface{} {
+	var params []interface{}
+	err := json.Unmarshal(req.Params, &params)
 	if err != nil {
 		s.writeError(w, err)
-		return
-	}
-
-	block := types.Block{}
-	blockData, err := hex.DecodeString(req.Data)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-	err = block.UnmarshalBinary(blockData)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	res := s.mock.SubmitBlock(r.Context(), &block)
-	code := 0
-	if res.Code != da.StatusSuccess {
-		code = 3
+		return nil
 	}
+	return params
+}
 
-	resp, err := json.Marshal(cnc.TxResponse{
-		Height: int64(res.DAHeight),
-		Code:   uint32(code),
-		RawLog: res.Message,
-	})
+func (s *Server) returnResponse(w http.ResponseWriter, resp *response) {
+	bytes, err := json.Marshal(resp)
 	if err != nil {
 		s.writeError(w, err)
 		return
 	}
-
-	s.writeResponse(w, resp)
+	s.writeResponse(w, bytes)
 }
 
-func (s *Server) shares(w http.ResponseWriter, r *http.Request) {
-	height, err := parseHeight(r)
+func (s *Server) rpc(w http.ResponseWriter, r *http.Request) {
+	var req request
+	err := json.NewDecoder(r.Body).Decode(&req)
 	if err != nil {
 		s.writeError(w, err)
 		return
 	}
+	switch req.Method {
+	case "header.GetByHeight":
+		params := s.parseParams(w, req)
```

Comment:
returning from `parseParams` on error case will not return here. i think this will not work. may be defining `parseParams` was an overkill. let's inline the code so that we have a working functionality on error case.

Reply:
restored.